How-to-find-the-commit-which-introduced-a-bug.md

```diff
@@ -0,0 +1,181 @@
+
+
+This will display the blame information for each line directly on the GitHub website. You can see the commit hash, author, and date for each line in the file.
+
+![Screenshot of the commit log highlighted](images/findCommitWhichIntroducedBug/image8.png)
+
+
+If you click on the "Blame Info" as shown in the above image you will be forwarded to the commit page where it belongs.
+
+
+![Screenshot asking you to click on "blame info"](images/findCommitWhichIntroducedBug/image4.png)
+
+
+## Git Bisect
+
+**`git bisect` is a powerful Git command that helps you find the commit that introduced a bug or regression in your codebase. It uses a binary search algorithm to efficiently narrow down the range of commits where the issue was introduced. However, you should note that this method is more useful if you don't know which line of code is causing the error (otherwise, we recommend using the "git blame" approach above instead).**
+
+
+Steps to use Git Bisect:
+
+1. Open your Terminal.
+
+2. navigate to your git repository using the `cd` command.
+
+    ```bash
+    cd opensource/oppia
+    ```
+
+
+3. Start the “git bisect” process.
+
+    ```bash
+    git bisect start
+    ```
+
+
+4. Mark the current commit as bad (where the bug exists):
+
+    ```bash
+    git bisect bad
+    ```
+
+
+5. Mark an older commit known to be bug-free as good (see [point 2](#bookmark=kix.vtty5bx24m2x) in the initial instructions):
+
+    ```bash
+    git bisect good <commit-id>
+    ```
+
+
+6. **Git will automatically checkout a commit in the middle of the range. Test your code to determine if the bug is present (e.g. refresh your browser and try to reproduce the error).**
+    * If the bug is present, mark the current commit as bad, so that “git bisect” only searches the range from the good commit to the current commit:
+
+        ```bash
+        git bisect bad
+        ```
+
+
+    * If the bug is not present, mark the current commit as good, so that “git bisect” only searches the range from the current commit to the latest commit on develop:
+
+        ```bash
+        git bisect good
+        ```
+
+
+7. Repeat step 6 until Git identifies the commit where the bug was introduced.
+8. Once Git finds the first bad commit, it will display the commit hash.
+9. To end the “git bisect” process, reset back to your original branch:
+
+    ```bash
+    git bisect reset
+    ```
+
+
+    Now, you have the commit-id that has introduced the bug. Report it in the issue thread, and then you can either fix it forward or revert the problematic commit.
+
+
+    **Tip:** While bisecting, you might need to inspect the changes in each commit. You can use this command to view those details: 
+
+    ```bash
+    git show <commit-id>
+    ```
+
+
+## Appendix: Using “`git blame`” on CLI
+
+**If you prefer using your terminal, here is how you can do “`git blame`”:**
+
+i. First navigate the file location using the `cd` command.
```

Comment:
...navigate to the...

Also why is the numbering here different from above? Let's be consistent.

Reply:
Done

Reply:
![image](https://github.com/oppia/oppia-web-developer-docs/assets/112589278/6e7ee593-479f-4bd1-a6a2-d7ad3ca745dc)